selfservice/hook/web_hook.go

```diff
@@ -499,3 +504,23 @@ func isTimeoutError(err error) bool {
 	var te interface{ Timeout() bool }
 	return errors.As(err, &te) && te.Timeout() || errors.Is(err, context.DeadlineExceeded)
 }
+
+func instrumentHTTPClientForEvents(ctx context.Context, httpClient *retryablehttp.Client) {
+	var (
+		attempt   = 0
+		requestID uuid.UUID
+		reqBody   []byte
+	)
+	httpClient.RequestLogHook = func(_ retryablehttp.Logger, req *http.Request, retryNumber int) {
+		attempt = retryNumber + 1
+		requestID = uuid.Must(uuid.NewV4())
+		req.Header.Set("Ory-Webhook-Request-ID", requestID.String())
+		reqBody, _ = httputil.DumpRequestOut(req, true)
+	}
+	httpClient.ResponseLogHook = func(_ retryablehttp.Logger, res *http.Response) {
+		res.Body = io.NopCloser(io.LimitReader(res.Body, 5<<20)) // read at most 5 MB from the response
```

Comment:
This modifies the response itself, so it will also limit the body read in the webhook itself. Doesn't it make more sense to add a global limit there instead? Because this is only set when the `emitEvent` config is set correct?

Reply:
I've added the `io.LimitReader` also to the code path where no event is emitted. Please check if that makes sense.

This is the easiest way to force a "global" limit, I think